lnwire/custom_records.go

```diff
@@ -0,0 +1,176 @@
+package lnwire
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// MinCustomRecordsTlvType is the minimum custom records TLV type as
+	// defined in BOLT 01.
+	MinCustomRecordsTlvType = 65536
+)
+
+// CustomRecords stores a set of custom key/value pairs. Map keys are TLV types
+// which must be greater than or equal to MinCustomRecordsTlvType.
+type CustomRecords map[uint64][]byte
+
+// NewCustomRecords creates a new CustomRecords instance from a
+// tlv.TypeMap.
+func NewCustomRecords(tlvMap tlv.TypeMap) (CustomRecords, error) {
+	// Make comparisons in unit tests easy by returning nil if the map is
+	// empty.
+	if len(tlvMap) == 0 {
+		return nil, nil
```

Comment:
This makes a lot of the code where we actually use this much nicer. So we don't need to check for `len()` every time we create the custom records from the remaining parsed TLV fields.
It also simplifies some of the unit tests, in that encoding/decoding of an empty custom record produces the same result, which is `nil` (instead of being an empty map after decoding).

How strongly do you feel about this? Changing it will impact many commits in the mega-PR...

Reply:
just somewhat, but non-blocking, def don't wanna bring large rebase conflicts and I can always create a PR if I reeeeally feel that strong